engines/sci/graphics/text16.cpp

```diff
@@ -559,22 +560,56 @@ void GfxText16::Box(const char *text, uint16 languageSplitter, bool show, const
 			break;
 		Width(curTextLine, 0, charCount, fontId, textWidth, textHeight, true);
 		maxTextWidth = MAX<int16>(maxTextWidth, textWidth);
-		switch (alignment) {
-		case SCI_TEXT16_ALIGNMENT_RIGHT:
-			offset = rect.width() - textWidth;
-			break;
-		case SCI_TEXT16_ALIGNMENT_CENTER:
-			offset = (rect.width() - textWidth) / 2;
-			break;
-		case SCI_TEXT16_ALIGNMENT_LEFT:
-			offset = 0;
-			break;
+		if (!g_sci->isLanguageRTL()) {
+			switch (alignment) {
+			case SCI_TEXT16_ALIGNMENT_RIGHT:
+				offset = rect.width() - textWidth;
+				break;
+			case SCI_TEXT16_ALIGNMENT_CENTER:
+				offset = (rect.width() - textWidth) / 2;
+				break;
+			case SCI_TEXT16_ALIGNMENT_LEFT:
+				offset = 0;
+				break;
 
-		default:
-			warning("Invalid alignment %d used in TextBox()", alignment);
+			default:
+				warning("Invalid alignment %d used in TextBox()", alignment);
+			}
+		} else {
+			// language direction is from Right to Left
+			switch (alignment) {
+			case SCI_TEXT16_ALIGNMENT_LEFT:
+				offset = rect.width() - textWidth;
+				break;
+			case SCI_TEXT16_ALIGNMENT_CENTER:
+				offset = (rect.width() - textWidth) / 2;
+				break;
+			case SCI_TEXT16_ALIGNMENT_RIGHT:
+				offset = 0;
+				break;
+
+			default:
+				warning("Invalid alignment %d used in TextBox()", alignment);
+			}
```

Comment:
IMHO, there's no need to add a separate block here. You can test for text alignment in the SCI_TEXT16_ALIGNMENT_RIGHT and SCI_TEXT16_ALIGNMENT_LEFT cases in the original switch